src/aiortc/rtcpeerconnection.py

```diff
@@ -1056,6 +1061,18 @@ def __createDtlsTransport(self) -> RTCDtlsTransport:
         # create ICE transport
         iceGatherer = RTCIceGatherer(iceServers=self.__configuration.iceServers)
         iceGatherer.on("statechange", self.__updateIceGatheringState)
+        if (
+            self.__configuration.bundlePolicy == RTCBundlePolicy.BALANCED
+            and len(self.__transceivers) > 0
+        ):
+            # ICE credentials are shared in "balanced" mode.
+            # TODO: fix aioice to take parameters instead
```

Comment:
This is a hack obviously and we would need an update to aioice to construct a gatherer with local username and password

Note that "balanced" does not mean "shares a single gatherer", this is observable in Chrome which with the default bundle gathers different local candidates.

max-bundle actually shares a single gatherer

Reply:
@jlaine shall I do a PR for aioice? I am a bit torn since this is not in the ORTC spec and should not be necessary to share but this is how all browsers behave and the more aligned one is with them...